### PR TITLE
[FLINK-25739][dist] Include Changelog to flink-dist jar

### PR DESCRIPTION
**Pull request by rkhachatryan**
## What is the purpose of the change

Include Changelog jars into dist/opt.

## Verifying this change

This change is a trivial rework without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? no

### CODE REVIEW

**Comment by rkhachatryan** [comment]
CI failure unrelated: https://issues.apache.org/jira/browse/FLINK-25740

File: flink-dist/src/main/assemblies/opt.xml [hunk 1]
```diff
@@ -118,13 +118,28 @@
...
+        <file>
 			<source>target/temporary/flink-shaded-zookeeper-3-${zookeeper.optional.version}-${flink.shaded.version}.jar</source>
 			<outputDirectory>opt</outputDirectory>
 			<destName>flink-shaded-zookeeper-${zookeeper.optional.version}.jar</destName>
 			<fileMode>0644</fileMode>
 		</file>
-	</files>
+
+        <file>
+            <source>../flink-dstl/flink-dstl-dfs/target/flink-dstl-dfs-${project.version}.jar</source>
```
Comment (zentol) [inline flink-dist/src/main/assemblies/opt.xml:129]: you should add dependencies to flink-dist for both of these modules to ensure they are built before flink-dist.

**Comment by rkhachatryan** [comment]
I've updated PR to include both jars into the Flink jar as agreed offline. PTAL.

File: flink-dist/pom.xml [hunk 2]
```diff
@@ -108,6 +108,18 @@ under the License.
...
+            <groupId>org.apache.flink</groupId>
+            <artifactId>flink-statebackend-changelog</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+
+        <dependency>
+			<groupId>org.apache.flink</groupId>
+			<artifactId>flink-dstl-dfs</artifactId>
+			<version>${project.version}</version>
+        </dependency>
```
Comment (zentol) [inline flink-dist/pom.xml:121]: inconsistent indentation

**Review by zentol — APPROVED** [review]

File: flink-dist/pom.xml (hunk 2 above)
Comment (rkhachatryan) [reply flink-dist/pom.xml:121]: fixed in b14ababb8a5d1433710d69f2184b475c335dac2d

**Pull request merged**